src/defs.h

```diff
@@ -108,6 +112,26 @@ typedef enum {
 
 /* variable definition */
 typedef struct {
+    int counter;
+    int stack[64];
+    int stack_idx;
+} rename_t;
+
+struct basic_block_t;
+
+struct ref_block_t {
```

Comment:
No suffix `_t` is allowed in structure definition. Instead, suffix `_t` appears in the `typedef`. That is,
```c
struct ref_block {
    basic_block_t *bb;
    struct ref_block *next;
};

typedef struct ref_block ref_block_t;
```